### PR TITLE
[SPARK-34905][SQL][TESTS] Enable ANSI intervals in `SQLQueryTestSuite`/`ThriftServerQueryTestSuite`

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
Remove `spark.sql.legacy.interval.enabled` settings from `SQLQueryTestSuite`/`ThriftServerQueryTestSuite` that enables new ANSI intervals by default.

### Why are the changes needed?
To use default settings for intervals, and test new ANSI intervals - year-month and day-time interval introduced by SPARK-27793.

### Does this PR introduce _any_ user-facing change?
Should not because this affects tests only.

### How was this patch tested?
By running the affected tests, for instance:
```
$ build/sbt "sql/testOnly *SQLQueryTestSuite -- -z datetime.sql"
$ build/sbt "sql/testOnly *SQLQueryTestSuite -- -z date.sql"
$ build/sbt "sql/testOnly *SQLQueryTestSuite -- -z timestamp.sql"
$ build/sbt "sql/testOnly *SQLQueryTestSuite -- -z interval.sql"
```

### CODE REVIEW

**Comment by MaxGekk** [comment]
Waiting for https://github.com/apache/spark/pull/32120 and https://github.com/apache/spark/pull/32121

**Comment by MaxGekk** [comment]
@yaooqinn @AngersZhuuuu @HyukjinKwon @cloud-fan Could you review this PR, please.

File: sql/core/src/test/resources/sql-tests/results/ansi/datetime.sql.out [hunk 1]
```diff
@@ -359,31 +359,31 @@ cannot resolve '1 + (- INTERVAL '2 seconds')' due to data type mismatch: argumen
 -- !query
 select date'2020-01-01' - timestamp'2019-10-06 10:11:12.345678'
 -- !query schema
-struct<subtracttimestamps(DATE '2020-01-01', TIMESTAMP '2019-10-06 10:11:12.345678'):interval>
+struct<subtracttimestamps(DATE '2020-01-01', TIMESTAMP '2019-10-06 10:11:12.345678'):day-time interval>
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/results/ansi/datetime.sql.out:362]: Have we finalized the SQL name for the new interval types? How about other databases?

Comment (MaxGekk) [reply sql/core/src/test/resources/sql-tests/results/ansi/datetime.sql.out:362]: Not yet. So far, I took the names for the sub-types from the SQL standard, see https://github.com/apache/spark/pull/31810

Probably, we will need to re-define them when we will implement parsing of interval types from SQL.

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**